core/interface.go

```diff
@@ -0,0 +1,21 @@
+package core
+
+type Extractor interface {
```

Comment:
Let's extend this interface to also include a description of the extractor. 
`Description ` as well as `Name` The same name can then be used in the recipe when defining source or sink etc.

Reply:
Where are we going to print this description?